index.js

```diff
@@ -119,14 +111,46 @@
   //  $$type :: String
   var $$type = '@@type';
 
-  //  type :: Any -> String
+  //  RPARSE :: RegExp
+  var RPARSE = /^(?:([^/\n\r]+)\/)?(.+?)(?:@(\d+))?$/;
+
+  //       TypeIdentifier :: (a, b, c) -> TypeIdentifier a b c
```

Comment:
:zap: 

Now this proposed signature shows that you expect `version` to be a String, whereas I modeled the code to expect an integer. I think it's nice to guarantee that one can do:

```js
type(x).version < type(y).version
```

But maybe that's taking too much responsibility.